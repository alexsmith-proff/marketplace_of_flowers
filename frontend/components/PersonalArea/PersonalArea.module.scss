.title {
    color: #222A46;
    font-size: 30px;
    font-style: normal;
    font-weight: 700;
    line-height: 40px;
    margin-bottom: 20px;
}

.table {
    width: 100%;
    border-radius: 10px;
    border: 1px solid #F2F3F5;
    box-shadow: 0px 4px 15px 0px rgba(34, 42, 70, 0.08);
    border-collapse: collapse;
}


.th {
    color: #505567;
    font-size: 13px;
    line-height: 20px;
    padding: 15px 0;
    background-color: #F4F5F6;

    &:nth-child(1) {
        width: 12%;
    }

    &:nth-child(2) {
        width: 18%;
    }

    &:nth-child(3) {
        width: 10%;
    }

    &:nth-child(4) {
        width: 15%;
    }

    &:nth-child(5) {
        width: 10%;
    }

    &:nth-child(6) {
        width: 7%;
    }
}

.td {
    padding: 20px 0;
    &.awaitingpayment {
        color: #2ad62a;
    }
    
    &.canceled {
        font-weight: bold;
    }
    
    &:nth-child(1) {
        text-align: center;
    }

    &:nth-child(3) {
        text-align: center;
    }
    
    &:nth-child(4) {
        text-align: center;
    }
    
    &:nth-child(5) {
        font-size: 15px;
        text-align: center;
    }

    &:nth-child(6) {
        text-align: center;
    }
}

.tr{
    border-bottom: 1px solid #E2E4EB;
    &:last-child{
        border-bottom: none;
    } 
}