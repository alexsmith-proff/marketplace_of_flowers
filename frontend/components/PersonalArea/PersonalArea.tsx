import { FC } from "react"

import s from './PersonalArea.module.scss'
import { useSelector } from "react-redux"
import { RootState } from "../../redux/store"

interface PersonalAreaProps {
}

const PersonalArea: FC<PersonalAreaProps> = ({ }) => {
    const userProfile = useSelector((orders: RootState) => orders.user.profile)

    console.log(userProfile);

    return (
        <div>
            <div className={s.title}>Личный кабинет</div>
            <table className={s.table}>
                <thead className={s.thead}>
                    <tr>
                        <th className={s.th}>Номер заказа</th>
                        <th className={s.th}>Название</th>
                        <th className={s.th}>Дата доставки</th>
                        <th className={s.th}>Получатель</th>
                        <th className={s.th}>Статус</th>
                        <th className={s.th}>Сумма</th>
                    </tr>
                </thead>
                <tbody>
                    {
                        userProfile.orders?.map(order => (
                            <tr className={s.tr} key={order.id}>
                                <td className={s.td}>{order.number}</td>
                                <td className={s.td}>{order.name}</td>
                                <td className={s.td}>{order.deliveryDate}</td>
                                <td className={s.td}>{userProfile.name}</td>
                                <td className={s.td + ' ' + s.awaitingpayment}>{order.status}</td>
                                <td className={s.td}>{order.price} ₽</td>
                            </tr>
                        ))
                    }
                </tbody>
            </table>

        </div>
    )
}

export default PersonalArea