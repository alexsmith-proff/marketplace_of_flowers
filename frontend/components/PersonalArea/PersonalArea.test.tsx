import { render, screen } from "@testing-library/react"
import '@testing-library/jest-dom/extend-expect'
import PersonalArea from "./PersonalArea"
import * as reduxHooks from "react-redux"
import { IUser } from "../../interfaces/users.interface"
import { UserOrdersStatus } from "../../enums/User.enum"

jest.mock('react-redux')
const mockedUseSelector = jest.spyOn(reduxHooks, 'useSelector')

const userProfileNoOrders: IUser = {
    profile: {
        id: 1,
        name: 'myName',
        email: 'name@mail.ru',
        orders: []
    },
    isVisibleAuthForm: false
}
const userProfileOrders: IUser = {
    profile: {
        id: 1,
        name: 'myName',
        email: 'name@mail.ru',
        orders: [
            {
                id: 1,
                name: 'Букет цветов',
                number: 5,
                price: 15000,
                status: UserOrdersStatus.PAID
            },
            {
                id: 2,
                name: 'Букет тюльпанов',
                number: 3,
                price: 10500,
                status: UserOrdersStatus.AWAITINGPAYMENT
            }
        ]
    },
    isVisibleAuthForm: false
}

describe('PersonalArea component', () => {
    it('PersonalArea renders without data', () => {
        mockedUseSelector.mockReturnValue(userProfileNoOrders.profile)
        render(<PersonalArea />)
        expect(screen.getByText(/Личный кабинет/i)).toBeInTheDocument()
        expect(screen.queryByText(/Букет тюльпанов/i)).toBeNull()
    })
    it('PersonalArea renders with data', () => {
        mockedUseSelector.mockReturnValue(userProfileOrders.profile)
        render(<PersonalArea />)
        expect(screen.getByText(/Букет тюльпанов/i)).toBeInTheDocument()
    })
    it('PersonalArea empty snapshot', () => {
        mockedUseSelector.mockReturnValue(userProfileNoOrders.profile)
        const component = render(<PersonalArea />)
        expect(component).toMatchSnapshot()
    })
    it('PersonalArea snapshot', () => {
        mockedUseSelector.mockReturnValue(userProfileOrders.profile)
        const component = render(<PersonalArea />)
        expect(component).toMatchSnapshot()
    })
})