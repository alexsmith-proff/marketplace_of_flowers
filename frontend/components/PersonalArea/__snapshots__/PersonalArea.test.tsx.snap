// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`PersonalArea component PersonalArea empty snapshot 1`] = `
{
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div>
        <div
          class="title"
        >
          Личный кабинет
        </div>
        <table
          class="table"
        >
          <thead
            class="thead"
          >
            <tr>
              <th
                class="th"
              >
                Номер заказа
              </th>
              <th
                class="th"
              >
                Название
              </th>
              <th
                class="th"
              >
                Дата доставки
              </th>
              <th
                class="th"
              >
                Получатель
              </th>
              <th
                class="th"
              >
                Статус
              </th>
              <th
                class="th"
              >
                Сумма
              </th>
            </tr>
          </thead>
          <tbody />
        </table>
      </div>
    </div>
  </body>,
  "container": <div>
    <div>
      <div
        class="title"
      >
        Личный кабинет
      </div>
      <table
        class="table"
      >
        <thead
          class="thead"
        >
          <tr>
            <th
              class="th"
            >
              Номер заказа
            </th>
            <th
              class="th"
            >
              Название
            </th>
            <th
              class="th"
            >
              Дата доставки
            </th>
            <th
              class="th"
            >
              Получатель
            </th>
            <th
              class="th"
            >
              Статус
            </th>
            <th
              class="th"
            >
              Сумма
            </th>
          </tr>
        </thead>
        <tbody />
      </table>
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;

exports[`PersonalArea component PersonalArea snapshot 1`] = `
{
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div>
        <div
          class="title"
        >
          Личный кабинет
        </div>
        <table
          class="table"
        >
          <thead
            class="thead"
          >
            <tr>
              <th
                class="th"
              >
                Номер заказа
              </th>
              <th
                class="th"
              >
                Название
              </th>
              <th
                class="th"
              >
                Дата доставки
              </th>
              <th
                class="th"
              >
                Получатель
              </th>
              <th
                class="th"
              >
                Статус
              </th>
              <th
                class="th"
              >
                Сумма
              </th>
            </tr>
          </thead>
          <tbody>
            <tr
              class="tr"
            >
              <td
                class="td"
              >
                5
              </td>
              <td
                class="td"
              >
                Букет цветов
              </td>
              <td
                class="td"
              />
              <td
                class="td"
              >
                myName
              </td>
              <td
                class="td awaitingpayment"
              >
                Оплачено
              </td>
              <td
                class="td"
              >
                15000
                 ₽
              </td>
            </tr>
            <tr
              class="tr"
            >
              <td
                class="td"
              >
                3
              </td>
              <td
                class="td"
              >
                Букет тюльпанов
              </td>
              <td
                class="td"
              />
              <td
                class="td"
              >
                myName
              </td>
              <td
                class="td awaitingpayment"
              >
                Ожидает оплаты
              </td>
              <td
                class="td"
              >
                10500
                 ₽
              </td>
            </tr>
          </tbody>
        </table>
      </div>
    </div>
  </body>,
  "container": <div>
    <div>
      <div
        class="title"
      >
        Личный кабинет
      </div>
      <table
        class="table"
      >
        <thead
          class="thead"
        >
          <tr>
            <th
              class="th"
            >
              Номер заказа
            </th>
            <th
              class="th"
            >
              Название
            </th>
            <th
              class="th"
            >
              Дата доставки
            </th>
            <th
              class="th"
            >
              Получатель
            </th>
            <th
              class="th"
            >
              Статус
            </th>
            <th
              class="th"
            >
              Сумма
            </th>
          </tr>
        </thead>
        <tbody>
          <tr
            class="tr"
          >
            <td
              class="td"
            >
              5
            </td>
            <td
              class="td"
            >
              Букет цветов
            </td>
            <td
              class="td"
            />
            <td
              class="td"
            >
              myName
            </td>
            <td
              class="td awaitingpayment"
            >
              Оплачено
            </td>
            <td
              class="td"
            >
              15000
               ₽
            </td>
          </tr>
          <tr
            class="tr"
          >
            <td
              class="td"
            >
              3
            </td>
            <td
              class="td"
            >
              Букет тюльпанов
            </td>
            <td
              class="td"
            />
            <td
              class="td"
            >
              myName
            </td>
            <td
              class="td awaitingpayment"
            >
              Ожидает оплаты
            </td>
            <td
              class="td"
            >
              10500
               ₽
            </td>
          </tr>
        </tbody>
      </table>
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;

exports[`PersonalArea component render FormHead 1`] = `
{
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div>
        <div
          class="title"
        >
          Личный кабинет
        </div>
        <table
          class="table"
        >
          <thead
            class="thead"
          >
            <tr>
              <th
                class="th"
              >
                Номер заказа
              </th>
              <th
                class="th"
              >
                Название
              </th>
              <th
                class="th"
              >
                Дата доставки
              </th>
              <th
                class="th"
              >
                Получатель
              </th>
              <th
                class="th"
              >
                Статус
              </th>
              <th
                class="th"
              >
                Сумма
              </th>
            </tr>
          </thead>
          <tbody />
        </table>
      </div>
    </div>
  </body>,
  "container": <div>
    <div>
      <div
        class="title"
      >
        Личный кабинет
      </div>
      <table
        class="table"
      >
        <thead
          class="thead"
        >
          <tr>
            <th
              class="th"
            >
              Номер заказа
            </th>
            <th
              class="th"
            >
              Название
            </th>
            <th
              class="th"
            >
              Дата доставки
            </th>
            <th
              class="th"
            >
              Получатель
            </th>
            <th
              class="th"
            >
              Статус
            </th>
            <th
              class="th"
            >
              Сумма
            </th>
          </tr>
        </thead>
        <tbody />
      </table>
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;

exports[`PersonalArea render FormHead 1`] = `
{
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div>
        <div
          class="title"
        >
          Личный кабинет
        </div>
        <table
          class="table"
        >
          <thead
            class="thead"
          >
            <tr>
              <th
                class="th"
              >
                Номер заказа
              </th>
              <th
                class="th"
              >
                Название
              </th>
              <th
                class="th"
              >
                Дата доставки
              </th>
              <th
                class="th"
              >
                Получатель
              </th>
              <th
                class="th"
              >
                Статус
              </th>
              <th
                class="th"
              >
                Сумма
              </th>
            </tr>
          </thead>
          <tbody />
        </table>
      </div>
    </div>
  </body>,
  "container": <div>
    <div>
      <div
        class="title"
      >
        Личный кабинет
      </div>
      <table
        class="table"
      >
        <thead
          class="thead"
        >
          <tr>
            <th
              class="th"
            >
              Номер заказа
            </th>
            <th
              class="th"
            >
              Название
            </th>
            <th
              class="th"
            >
              Дата доставки
            </th>
            <th
              class="th"
            >
              Получатель
            </th>
            <th
              class="th"
            >
              Статус
            </th>
            <th
              class="th"
            >
              Сумма
            </th>
          </tr>
        </thead>
        <tbody />
      </table>
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;

exports[`PersonalArea render orders 1`] = `
{
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div>
        <div
          class="title"
        >
          Личный кабинет
        </div>
        <table
          class="table"
        >
          <thead
            class="thead"
          >
            <tr>
              <th
                class="th"
              >
                Номер заказа
              </th>
              <th
                class="th"
              >
                Название
              </th>
              <th
                class="th"
              >
                Дата доставки
              </th>
              <th
                class="th"
              >
                Получатель
              </th>
              <th
                class="th"
              >
                Статус
              </th>
              <th
                class="th"
              >
                Сумма
              </th>
            </tr>
          </thead>
          <tbody />
        </table>
      </div>
    </div>
  </body>,
  "container": <div>
    <div>
      <div
        class="title"
      >
        Личный кабинет
      </div>
      <table
        class="table"
      >
        <thead
          class="thead"
        >
          <tr>
            <th
              class="th"
            >
              Номер заказа
            </th>
            <th
              class="th"
            >
              Название
            </th>
            <th
              class="th"
            >
              Дата доставки
            </th>
            <th
              class="th"
            >
              Получатель
            </th>
            <th
              class="th"
            >
              Статус
            </th>
            <th
              class="th"
            >
              Сумма
            </th>
          </tr>
        </thead>
        <tbody />
      </table>
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
